pkg/kstatus/status/core.go

```diff
@@ -262,8 +264,14 @@ func replicasetConditions(u *unstructured.Unstructured) (*Result, error) {
 	for _, c := range objc.Status.Conditions {
 		// https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/pkg/controller/replicaset/replica_set_utils.go
 		if c.Type == "ReplicaFailure" && c.Status == corev1.ConditionTrue {
+			reason := "ReplicaFailure"
 			message := "Replica Failure condition. Check Pods"
-			return newInProgressStatus("ReplicaFailure", message), nil
```

Comment:
Do we ever call this function without the `lastTransitionTime`? It seems like it should always be set, so should we change the function signature to accept a timestamp?

Reply:
`newInProgressStatus` and other `newXXXStatus` should set the `LastTransitionTime` to current time.
Overall, `newXXXStatus` are called much more frequently than returning a `Result` whose `LastTransitionTime` is from  an existing condition in an `unstructured.Unstructured` object.